hs-bindgen/src-internal/HsBindgen/Frontend/Pass/MangleNames.hs

```diff
@@ -59,9 +61,25 @@ mangleNames unit =
 type NameMap = Map (C.QualId HandleTypedefs) HsIdentifier
 
 data MangleError =
-    CouldNotMangle Text
+    CouldNotMangle !Text
+  | MissingDeclaration !(C.QualId HandleTypedefs)
   deriving stock (Show, Eq)
 
+instance PrettyTrace MangleError where
+  prettyTrace (CouldNotMangle name) =
+    "Could not mangle C name: " <> unpack name
+  -- NB: We did not register any declaration with the given ID. This is most
+  -- likely because the user did not select the declaration. If the declaration
+  -- was completely missing, Clang would have complained already.
+  prettyTrace (MissingDeclaration qualId) =
```

Comment:
Let's have this comment where we generate the message, rather than where we pretty-print it.